tests/a_scripts/inference/test_convert_composer_to_hf.py

```diff
@@ -590,23 +627,54 @@ def test_huggingface_conversion_callback(
                 name for name in os.listdir(
                     os.path.join(tmp_path, 'checkpoints', 'huggingface'))
             ]
+
+            checkpoint_files = os.listdir(
+                os.path.join(tmp_path, 'checkpoints', 'huggingface',
+                             huggingface_checkpoints[-1]))
+            if peft_config is not None:
+                assert 'adapter_config.json' in checkpoint_files
+                assert 'adapter_model.safetensors' in checkpoint_files
+
             assert len(normal_checkpoints) == expected_normal_checkpoints
             assert len(huggingface_checkpoints) == expected_hf_checkpoints
 
             # Patch flash_attn package to be empty to simulate loading the model in
-            # an environment without flash atttention installed
+            # an environment without flash attention installed
             with patch.dict('sys.modules', {'flash_attn': None}):
+                if peft_config is not None:
+                    composer_model = trainer.state.model.module if trainer.state.is_model_ddp else trainer.state.model
+                    composer_model.model.base_model.save_pretrained(
+                        tmp_path / 'base-model')
+
+                checkpoint_path = os.path.join(tmp_path, 'checkpoints',
+                                               'huggingface',
+                                               f'ba{batches_per_epoch}')
+
+                if peft_config is not None:
+                    with open(
+                            os.path.join(checkpoint_path,
+                                         'adapter_config.json')) as _f:
+                        adapter_config = json.load(_f)
+
+                    adapter_config['base_model_name_or_path'] = str(
+                        tmp_path / 'base-model')
+
+                    with open(
+                            os.path.join(checkpoint_path,
+                                         'adapter_config.json'), 'w') as _f:
+                        json.dump(adapter_config, _f)
+
                 # Load the last huggingface checkpoint
                 loaded_model = transformers.AutoModelForCausalLM.from_pretrained(
-                    os.path.join(tmp_path, 'checkpoints', 'huggingface',
-                                 f'ba{batches_per_epoch}'),
+                    checkpoint_path,
                     trust_remote_code=True,
                 )
 
             # Check that the loaded model has the correct precision, and then set it back
             # to the original for the equivalence check
-            assert loaded_model.config.torch_dtype == precision
-            loaded_model.config.torch_dtype = original_model.model.config.torch_dtype
+            if peft_config is None:
```

Comment:
do we need to do a precision check for the peft_config is not None case?

Reply:
The config is the base model's config, and afaict the peft config doesn't have a precision option, so no, this is just for when not using peft.